docs/DeveloperGuide.adoc

```diff
@@ -767,6 +808,139 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Show walkthrough
+
+*MSS*
+
+1.  User requests to list commands.
+2.  AddressBook shows a list of commands and their input requirements.
++
+Use case ends.
+
+[discrete]
+=== Use case: Export AddressBook
+
+*MSS*
+
+1.  User requests to create a backup
+2.  AddressBook writes existing persons in the AddressBook into a file determined by the user
+
+[none]
+* 2a. Destination file already exist.
++
+[none]
+** 2a1. AddressBook overwrites the existing file.
++
+Use case ends.
+
+[discrete]
+=== Use case: Copy emails from list to clipboard
+
+*MSS*
+
+1. User request to list persons
+2. AddressBook shows a list of persons
+3. User requests to copy all or selected person's email in the list
+4. Emails are copied to clipboard
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Add person
+
+*MSS*
+
+1.  User requests to add person
+2.  AddressBook parses person's particulars into a fixed format
+3.  AddressBook adds the person
```

Comment:
usecase should record the back and forth interaction between the user and the system. this one violates it